language/move-vm/runtime/src/runtime.rs

```diff
@@ -377,19 +377,22 @@ impl VMRuntime {
         for (idx, (arg, arg_type)) in args.into_iter().zip(params).enumerate() {
             match arg_type {
                 Type::MutableReference(inner_t) => {
-                    dummy_locals
-                        .store_loc(idx, self.deserialize_value(&inner_t, arg).unwrap())
-                        .unwrap();
-                    actuals.push(dummy_locals.borrow_loc(idx).unwrap());
+                    if let Err(err) = self.arg_to_actual(idx, arg, &inner_t, &mut dummy_locals, &mut actuals) {
```

Comment:
Can't one use `self.arg_to_actual(...).map_err(..)?` or something?

Reply:
`map_err` can be used here, i tried reworking it a bit to use it but it didn't seem any shorter or more readable. Maybe there is something i'm missing in its use, do you have something specific in mind ?